setup.py

```diff
@@ -53,7 +53,8 @@ def maybe_require(pkg):
 client = set([
     'requests',
     'python-gnupg==0.4.6',
-    'oyaml'
+    'oyaml',
+    'setuptools; python_version>="3.12"',
```

Comment:
Hi @m-horky - Would you mind moving it to the `runtime` set? As the `insights-core` can also be installed via `pip` and it provides some entry-points for users to run rules or do analysis on their localhost. E.g. https://github.com/RedHatInsights/insights-core/blob/e361b98df12483270cbeb22544b328e44a822a18/setup.py#L12-L24 

And the same error will also occur in this case if py312 is used in the local env.